From stern@rowland.harvard.edu Thu Mar 27 07:15:30 2008
From: Alan Stern <stern@rowland.harvard.edu>
Date: Thu, 27 Mar 2008 10:15:22 -0400 (EDT)
Subject: USB: fix bug in sg initialization in usbtest
To: Greg KH <greg@kroah.com>
Cc: Robert Jarzmik <rjarzmik@free.fr>, David Brownell <david-b@pacbell.net>,  Jens Axboe <axboe@kernel.dk>
Message-ID: <Pine.LNX.4.44L0.0803271011440.3808-100000@iolanthe.rowland.org>


This patch (as1062) fixes a bug in the scatter-gather initialization
code in the usbtest driver.  When the sg-helper conversion was
performed, it wasn't done correctly.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
CC: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/usbtest.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -378,6 +378,7 @@ alloc_sglist (int nents, int max, int va
 	sg = kmalloc (nents * sizeof *sg, GFP_KERNEL);
 	if (!sg)
 		return NULL;
+	sg_init_table(sg, nents);
 
 	for (i = 0; i < nents; i++) {
 		char		*buf;
@@ -390,7 +391,7 @@ alloc_sglist (int nents, int max, int va
 		}
 
 		/* kmalloc pages are always physically contiguous! */
-		sg_init_one(&sg[i], buf, size);
+		sg_set_buf(&sg[i], buf, size);
 
 		switch (pattern) {
 		case 0:
